File: play-sound.mjs
#!/usr/bin/env node
import { execFileSync } from "node:child_process";
import { readFileSync } from "node:fs";

function isWSL() {
  if (process.env.WSL_DISTRO_NAME) return true;
  try {
    return readFileSync("/proc/version", "utf8").toLowerCase().includes("microsoft");
  } catch {
    return false;
  }
}

// 実行環境のキー（WSLはWindows側に投げる）
function platformKey() {
  if (process.platform === "linux" && isWSL()) return "wsl";
  return process.platform; // darwin / win32
}

// イベント名 → 環境ごとの [コマンド, 引数]
const SOUNDS = {
  darwin: {
    notification: ["afplay", ["/System/Library/Sounds/Glass.aiff"]],
    stop: ["afplay", ["/System/Library/Sounds/Funk.aiff"]],
  },
  win32: {
    notification: ["powershell", ["-c", "[console]::beep(880,200)"]],
    stop: ["powershell", ["-c", "[console]::beep(440,300)"]],
  },
  wsl: {
    // WSLからは powershell.exe を直接呼べる（Windows側で鳴る）
    notification: ["powershell.exe", ["-c", "[console]::beep(880,200)"]],
    stop: ["powershell.exe", ["-c", "[console]::beep(440,300)"]],
  },
};

export default function play(event = "notification") {
  try {
    const table = SOUNDS[platformKey()];
    if (!table) return;                     // 対象外の環境は静かに終了
    const [cmd, args] = table[event] ?? table.notification;
    execFileSync(cmd, args, { stdio: "ignore" });
  } catch {
    // プレイヤーが無くても落とさない
  }
}

File: play-sound.py
#!/usr/bin/env python3
import os
import sys
import subprocess


def _is_wsl():
    if os.environ.get("WSL_DISTRO_NAME"):
        return True
    try:
        return "microsoft" in open("/proc/version").read().lower()
    except OSError:
        return False


def _platform_key():
    if sys.platform == "linux" and _is_wsl():
        return "wsl"
    return sys.platform  # darwin / win32


def play(event="notification"):
    key = _platform_key()
    try:
        if key == "darwin":
            sounds = {
                "notification": "/System/Library/Sounds/Glass.aiff",
                "stop":         "/System/Library/Sounds/Funk.aiff",
            }
            subprocess.run(
                ["afplay", sounds.get(event, sounds["notification"])],
                stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL,
            )
        elif key == "win32":
            import winsound
            aliases = {
                "notification": "SystemAsterisk",
                "stop":         "SystemExclamation",
            }
            winsound.PlaySound(
                aliases.get(event, "SystemAsterisk"), winsound.SND_ALIAS
            )
        elif key == "wsl":
            # [System.Media.SystemSounds].Play() は非同期なので Sleep で完奏を待つ
            cmds = {
                "notification": "[System.Media.SystemSounds]::Asterisk.Play(); Start-Sleep -Milliseconds 800",
                "stop":         "[System.Media.SystemSounds]::Exclamation.Play(); Start-Sleep -Milliseconds 800",
            }
            subprocess.run(
                ["powershell.exe", "-c", cmds.get(event, cmds["notification"])],
                stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL,
            )
    except Exception:
        pass


if __name__ == "__main__":
    play(sys.argv[1] if len(sys.argv) > 1 else "notification")
